test/functional/config.js

```diff
@@ -22,6 +28,106 @@ module.exports = {
         },
         {
             platform:    'Windows 10',
+            browserName: 'firefox',
+            alias:       'firefox'
+        },
+        {
+            platform:    'OS X 10.11',
+            browserName: 'safari',
+            version:     '9.0',
+            alias:       'safari'
+        },
+        {
+            platform:    'OS X 10.11',
+            browserName: 'chrome',
+            alias:       'chrome-osx'
+        },
+        {
+            platform:    'OS X 10.11',
+            browserName: 'firefox',
+            alias:       'firefox-osx'
+        }
+    ]
+};
+
+testingEnvironments[testingEnvironmentNames.msDesktopBrowsers] = {
+    sauceLabs: {
+        username:  process.env.SAUCE_USERNAME_FUNCTIONAL_MS_DESKTOP,
+        accessKey: process.env.SAUCE_ACCESS_KEY_FUNCTIONAL_MS_DESKTOP,
+        jobName:   'functional tests - ms desktop browsers'
+    },
+
+    browsers: [
+        {
+            platform:    'Windows 10',
+            browserName: 'microsoftedge',
+            alias:       'edge'
+        },
+        {
+            platform:    'Windows 10',
+            browserName: 'internet explorer',
+            version:     '11.0',
+            alias:       'ie 11'
+        },
+        {
+            platform:    'Windows 8',
+            browserName: 'internet explorer',
+            version:     '10.0',
+            alias:       'ie 10'
+        },
+        {
+            platform:    'Windows 7',
+            browserName: 'internet explorer',
+            version:     '9.0',
+            alias:       'ie 9'
+        }
+    ]
+};
+
+testingEnvironments[testingEnvironmentNames.mobileBrowsers] = {
+    sauceLabs: {
+        username:  process.env.SAUCE_USERNAME_FUNCTIONAL_MOBILE,
+        accessKey: process.env.SAUCE_ACCESS_KEY_FUNCTIONAL_MOBILE,
+        jobName:   'functional tests - mobile browsers'
+    },
+
+    browsers: [
+        {
+            platformName:    'Android',
+            deviceName:      'Android Emulator',
+            platformVersion: '5.1',
+            browserName:     'Browser',
+            alias:           'android'
+        },
+        {
+            // NOTE: we can't run tests on iOS 9.3 because the bug in this version
+            // (see https://github.com/DevExpress/testcafe-hammerhead/issues/672#issuecomment-232043366).
+            // The bug is fixed in iOS 9.3.2 but it's unavailable on the farm.
```

Comment:
```
            // NOTE: we can't run tests on iOS 9.3 because of a bug in this version
            // (see https://github.com/DevExpress/testcafe-hammerhead/issues/672#issuecomment-232043366).
            // This bug is fixed in iOS 9.3.2 but it's not available on the farm.
```